### ISSUE TITLE
AsnFormatter.BinaryToString array overflow error

### ISSUE THREAD

**Issue by jalliot** [Crypt32/Asn1DerParser.NET#1]
Hello again,

After reading your last blog post on the AsnFormatter features, I tried to replace one of my functions that exported a certificate in a base64 file with your implementation.
It works perfectly on my own workstation with Windows 8.1 and PowerShell v5 but it fails on one of my colleagues' PC which is stuck on Windows 7 and PS v4 (I don't know if it is relevant but this is the only difference I see between our configurations...). Note that we both have the 3.2.6 release of PSPKI module installed, and so the same version of the ASN1 library.

When executing the following command on his PC:
``` powershell
[SysadminsLV.Asn1Parser.AsnFormatter]::BinaryToString($cert.RawData, [SysadminsLV.Asn1Parser.EncodingType]::Base64Header)
```
... he gets the following `IndexOutOfRangeException`:
```
Index was outside the bounds of the array.
```

For information, here is his `$PSVersionTable` (which shows .NET version):
```
Name                           Value
----                           -----
PSVersion                      4.0
WSManStackVersion              3.0
SerializationVersion           1.1.0.1
CLRVersion                     4.0.30319.34209  
BuildVersion                   6.3.9600.16406  
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0}
PSRemotingProtocolVersion      2.2
```

You can also have a look at the [CliXml export](https://github.com/Crypt32/Asn1DerParser.NET/files/415040/error.zip) of the error record which contains the stacktrace and so on.

For the time being, I've rollbacked to my own code which was working fine

**Comment by Crypt32** [comment]
Hi, I was able to repro the issue and now I'm working on it.

**Comment by Crypt32** [comment]
So, I figured out this issue. It is not related to `BinaryToString` method call it is a bug in previous versions of PowerShell: v3.0 and v4.0 and was fixed in 5.0. There is no chance to get it fixed in previous versions of PowerShell, but there is a fix: explicitly specify method overload to call:

``` PowerShell
[SysadminsLV.Asn1Parser.AsnFormatter]::BinaryToString($cert.RawData, [SysadminsLV.Asn1Parser.EncodingType]::Base64Header,0,0,$false)
```

I just tested and it works in all PowerShell versions.
